webapp/src/ts/services/session.service.ts

```diff
@@ -23,10 +23,13 @@ export class SessionService {
 
   navigateToLogin() {
     console.warn('User must reauthenticate');
+    const userCtx = this.userCtx();
+    const username = userCtx && userCtx.name;
+    const usernameQueryParam = username ? `username=${username}&` : '';
     this.cookieService.delete(COOKIE_NAME, '/');
     this.userCtxCookieValue = undefined;
     this.document.location.href =
-      `/${this.location.dbName}/login?redirect=${encodeURIComponent(this.document.location.href)}`;
+      `/${this.location.dbName}/login?${usernameQueryParam}redirect=${encodeURIComponent(this.document.location.href)}`;
```

Comment:
As above, I think it would be nicer to refactor this to use URLSearchParams:
```suggestion
    const params = new URLSearchParams();
    params.append('redirect', this.document.location.href);
    const userCtx = this.userCtx();
    const username = userCtx && userCtx.name;
    if (username) {
      params.append('username', username);
    }

    this.cookieService.delete(COOKIE_NAME, '/');
    this.userCtxCookieValue = undefined;
    this.document.location.href = `/${this.location.dbName}/login?${params.toString()}`;
```

Reply:
Hi @dianabarsan! Thank you for reviewing this PR.
I have one question.
Do you think `this.document.location.href` should be encoded in this way `encodeURIComponent(this.document.location.href)` as before?

Reply:
To both questions, URLSearchParams does it's own encoding: https://url.spec.whatwg.org/#example-constructing-urlsearchparams . There are some differences compared to encodeURIComponent, though. We're already using URLSearchParams on the backend to redirect: https://github.com/medic/cht-core/blob/master/api/src/public/login/script.js#L156

Reply:
@dianabarsan I updated the code as you requested. Thanks.